homeassistant/components/irm_kmi/entity.py

```diff
@@ -0,0 +1,32 @@
+"""Base class shared among IRM KMI entities."""
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, IRM_KMI_NAME
+from .utils import preferred_language
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IrmKmiBaseEntity(CoordinatorEntity):
+    """Base methods for IRM KMI entities."""
+
+    _attr_attribution = (
+        "Weather data from the Royal Meteorological Institute of Belgium meteo.be"
+    )
+
+    def __init__(self, entry: ConfigEntry, name: str) -> None:
```

Comment:
what's the name?

Reply:
It's fine to pass it to a base entity, given that it will be reused when other platforms are added. You may want to change the hardcoded string to a translatable name by using entity name translations.

Reply:
Solved by this discussion: https://github.com/home-assistant/core/pull/144689#discussion_r2265313345